plugins/org.polarsys.capella.cybersecurity.sirius.analysis/src/org/polarsys/capella/cybersecurity/sirius/analysis/CybersecurityServices.java

```diff
@@ -158,11 +158,25 @@ public Threat createThreat(EObject any) {
     return null;
   }
 
-  public String getThreatLabel(Threat threat) {
-    String kind = threat.getKind().getName().toUpperCase();
+  public String getThreatLabel(Threat threat) { 
+    String kind = null;
+    if (threat.getKind() != null) {
+      kind = threat.getKind().getName().toUpperCase();
+    } else {

```

Comment:
Choosing another kind should not be done here, the label should be coherent with the model state.